### PR TITLE
Enable fmt::format for Date values

### PR DESCRIPTION
**Pull request by pramodsatya**
Add a formatter for Date class to support Date values with fmt::format.

### CODE REVIEW

File: velox/functions/lib/tests/DateTimeFormatterTest.cpp [hunk 1]
```diff
@@ -2192,4 +2200,21 @@ TEST_F(MysqlDateTimeTest, parseFractionOfSecond) {
   EXPECT_THROW(parse("9999999", "%f", false), VeloxUserError);
 }
 
+class FmtDateFormatterTest : public DateTimeFormatterTest {};
+
+TEST_F(FmtDateFormatterTest, fmtDateFormatter) {
```
Comment (mbasmanova) [inline velox/functions/lib/tests/DateTimeFormatterTest.cpp:2205]: Let's not mixup different formatting concepts and move this test somewhere else. Is there already a test file for Date type? If not, let's add DateTest.cpp with DateTest test class and add TEST_F(DateTest, format).

File: velox/type/Date.h [hunk 2]
```diff
@@ -102,6 +102,20 @@ std::string to_string(const ::facebook::velox::Date& ts);
...
+template <>
+struct fmt::formatter<facebook::velox::Date> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const facebook::velox::Date& d, FormatContext& ctx) {
+    auto x = std::to_string(d);
```
Comment (mbasmanova) [inline velox/type/Date.h:114]: this variable is not needed; just use std::to_string directly.

File: velox/type/tests/TypeTest.cpp [hunk 3]
```diff
@@ -251,6 +251,27 @@ TEST(TypeTest, parseStringToDate) {
...
 
+TEST(TypeTest, dateFormat) {
+  auto parseDate = [](const std::string& dateStr) {
+    Date returnDate;
+    parseTo(dateStr, returnDate);
+    return returnDate;
+  };
+
+  EXPECT_EQ(
+      fmt::format("{}", parseDate("2015-12-24")), std::string("2015-12-24"));
```
Comment (mbasmanova) [inline velox/type/tests/TypeTest.cpp:262]: Is it necessary to call std::string for "2015-12-24" ? If not, let's remove to make the test shorter and easier to read.

**Review by mbasmanova — APPROVED** [review]
@pramodsatya Looks good % a couple of nits.

**Review by aditi-pandit — COMMENTED** [review]
Thanks Pramod. This looks good.

**Pull request closed without merging**